op-chain-ops/genesis/helpers.go

```diff
@@ -107,3 +108,18 @@ func newHexBig(in uint64) *hexutil.Big {
 	hb := hexutil.Big(*b)
 	return &hb
 }
+
+func maybeSetupEcotone4788(config *DeployConfig, db *state.MemoryStateDB) (*state.MemoryStateDB, error) {
+	genesis := db.Genesis()
+
+	ecotoneTime := config.EcotoneTime(genesis.Timestamp)
+
+	if ecotoneTime != nil && *ecotoneTime <= genesis.Timestamp {
+		err := setup4788Contract(db)
```

Comment:
This will need to be replaced with execution of the upgrade transactions but is ok for now

Reply:
Agreed, let's work on an genesis upgrade-txs squash design together later today.